features/flows/elastic/src/main/java/org/opennms/netmgt/flows/elastic/thresholding/FlowThresholding.java

```diff
@@ -76,36 +82,20 @@
 
     private final IpInterfaceDao ipInterfaceDao;
 
-    private final Map<ApplicationKey, AtomicLong> applicationAccumulator;
-
-    private final Cache<ExporterKey, Session> sessions;
+    private final ConcurrentMap<ExporterKey, Session> sessions = Maps.newConcurrentMap();
 
     private long stepSizeMs = 0;
+    private long idleTimeoutMs = 15 * 60 * 1000;
 
     private Timer timer;
 
     public FlowThresholding(final ThresholdingService thresholdingService,
                             final CollectionAgentFactory collectionAgentFactory,
-                            final IpInterfaceDao ipInterfaceDao,
-                            final CacheConfig sessionCacheConfig) {
+                            final IpInterfaceDao ipInterfaceDao) {
         this.thresholdingService = Objects.requireNonNull(thresholdingService);
         this.collectionAgentFactory = Objects.requireNonNull(collectionAgentFactory);
 
         this.ipInterfaceDao = Objects.requireNonNull(ipInterfaceDao);
-
-        //noinspection unchecked
-        this.applicationAccumulator = Maps.newConcurrentMap();
-
-        //noinspection unchecked
-        this.sessions = new CacheBuilder<ExporterKey, Session>()
-                .withConfig(sessionCacheConfig)
-                .withCacheLoader(new CacheLoader<ExporterKey, Session>() {
-                    @Override
-                    public Session load(final ExporterKey key) throws Exception {
-                        throw new IllegalStateException();
-                    }
-                })
-                .build();
     }
 
     public long getStepSizeMs() {
```

Comment:
I think that `setStepSize` should be synchronized. Otherwise you may miss a running timer.

Reply:
It's just a setting. I don't care if a run is lost in this case. This is not intended to be changed after initial config.

Reply:
The implementation explicitly checks for a running timer. This seems to foresee that the method is called multiple time.

Reply:
True. But again, this is just called by explicit user interaction for configuration. Configuration is not expected to be thread-safe.